src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -834,6 +1112,96 @@ private void AddMetaKeyValues(int i, DataViewSchema.Annotations.Builder builder)
             }
 
             protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer) => _parent.GetGetterCore(input, iinfo, out disposer);
+
+            private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, string srcVariable, string dstVariable)
+            {
+                string opType;
+                OnnxNode murmurNode;
+
+                opType = "MurmurHash3";
+                string murmurOutput = ctx.AddIntermediateVariable(_dstTypes[iinfo], "MurmurOutput");
+                var srcType = _srcTypes[iinfo].GetItemType().RawType;
+
+                // Numeric input types are limited to those supported by the Onnxruntime MurmurHash operator, which currently only supports
+                // uints and ints. Thus, ulongs, longs, doubles, floats, and booleans are not supported.
+                if (srcType == typeof(ushort) || srcType == typeof(short) || srcType == typeof(sbyte) || srcType == typeof(byte))
+                {
+                    var castOutput = ctx.AddIntermediateVariable(NumberDataViewType.UInt32, "CastOutput", true);
+                    var castNode = ctx.CreateNode("Cast", srcVariable, castOutput, ctx.GetNodeName(opType), "");
+                    castNode.AddAttribute("to", NumberDataViewType.UInt32.RawType);
+                    murmurNode = ctx.CreateNode(opType, castOutput, murmurOutput, ctx.GetNodeName(opType), "com.microsoft");
+                }
+                else if (srcType == typeof(uint) || srcType == typeof(int) || srcType == typeof(ReadOnlyMemory<char>))
+                {
+                    murmurNode = ctx.CreateNode(opType, srcVariable, murmurOutput, ctx.GetNodeName(opType), "com.microsoft");
+                }
+                else
+                {
+                    return false;
+                }
+
+                murmurNode.AddAttribute("positive", 1);
+                var seed = _parent._columns[iinfo].Seed;
+                if (_parent._columns[iinfo].UseOrderedHashing)
+                    seed = Hashing.MurmurRound(seed, 0);
+                murmurNode.AddAttribute("seed", seed);
+
+                // masking is done via bitshifts, until bitwise AND is supported by Onnxruntime
+                opType = "BitShift";
+                string bitShiftOutput = ctx.AddIntermediateVariable(_dstTypes[iinfo], "BitShiftOutput");
+                var shiftValue = ctx.AddInitializer((ulong)(32 - _parent._columns[iinfo].NumberOfBits), false);
+                var shiftNode = ctx.CreateNode(opType, new[] { murmurOutput, shiftValue }, new[] { bitShiftOutput }, ctx.GetNodeName(opType), "");
+                shiftNode.AddAttribute("direction", "LEFT");
+
+                opType = "BitShift";
+                var outputVariable = dstVariable;
+                if(_srcTypes[iinfo].GetItemType() is KeyDataViewType)
+                    outputVariable = ctx.AddIntermediateVariable(_dstTypes[iinfo], "BitShiftOutput2");
```

Comment:
>outputVariable [](start = 20, length = 14)

Why do we need this intermediate variable for key types?

Reply:
+1 is only for key types, so the output changes

Reply:
See my other comment regarding this - the output of `HashingTransformer` is always a key type.

---
In reply to: [413957695](https://github.com/dotnet/machinelearning/pull/5013#discussion_r413957695) [](ancestors = 413957695)